### ISSUE TITLE
Define the "slot type"

### ISSUE THREAD

**Issue by ogarciarevett** [G7DAO/contracts#8]
Define the type of slot:

Example:

If we have "accessories" in the inventory, how many rings could I have "active" ? 

This should be an Argument for the Diamond init ?
To be more reusable? 
How many "slots" will be "double" renderable in the UI? Only the "boots", "shoulders", and "hands"? What about "wings"?
We will have wings?

**Comment by ogarciarevett** [comment]
As a result of this "spike" we have these slot types:

1. Equippable NFTs(Already in the PR, ERC20, ERC721, ERC1155)
2. Materials NFTs(Unequippable NFT,  could be in the PR as a "0"  value but actually will be better if we create a new type, I'll add this to part 2 of the Inventory)
3. Trophy NFTs(is actually just a badge but is crafted, I don't really know if we need to put it as a different type for the smart contract side, but the UI will need this for sure)

**Comment by zomglings** [comment]
I don't think these should be stored on-chain. This should be part of the slot metadata. We do need to support `slotUri` on-chain, but I would hesitate to implement much more than that.

**Comment by ogarciarevett** [comment]
@zomglings I agree but not in all, we don't need to define the entire metadata but at least the `type` as an Enum with the `slotUri` could be useful for any inventory. If we do this off-chain we're always requiring that the `slot type` must be stored and a UI app can't use this without a backend for example.

So, giving a `slot_type` has an optional argument to the slot has better reusability purposes. It's optional.

I'll close the issue with this, but if anyone has different thoughts on this, just left a comment over here 👇 

Or any new idea is also valid 😄
